src/backend/commands/self/reset_weapons.cpp

```diff
@@ -0,0 +1,21 @@
+#include "backend/command.hpp"
+#include "script_function.hpp"
+
+namespace big
+{
+	class reset_weapons : command
+	{
+		using command::command;
+
+		virtual void execute(const command_arguments&, const std::shared_ptr<command_context> ctx)
+		{
+			if (*g_pointers->m_gta.m_is_session_started && scripts::is_running("freemode"_J))
+			{
+				*scr_globals::should_reset_fm_weapons.as<PBOOL>() = TRUE;
+				scr_functions::give_owned_freemode_weapons.call<void>();
+			}

```

Comment:
Could we tie this into Persist Weapons somehow, that if it detects a weapons malignment, it runs this instead? Make that a user configurable thing. We also want to check for a runaway scenario, where it's stuck in a loop. (It keeps detecting weapons have changed, but running the scr func has no outcome, thus either stop what it's doing or warn the user)